quick_start/kusto_sample_app.py

```diff
@@ -0,0 +1,261 @@
+    return KustoConnectionStringBuilder.with_aad_device_authentication(cluster)
+
+
+def create_managed_identity_connection_string(cluster: str) -> KustoConnectionStringBuilder:
+    # Connect using the system or user provided managed identity (azure service only)
+    # Todo - Config (Optional): Managed identity client id if you are using a User Assigned Managed Id
+    client_id = os.environ.get("MANAGED_IDENTITY_CLIENT_ID")
+
+    if client_id is None:
+        return KustoConnectionStringBuilder.with_aad_managed_service_identity_authentication(cluster)
+    else:
+        return KustoConnectionStringBuilder.with_aad_managed_service_identity_authentication(cluster, client_id=client_id)
+
+
+def create_app_key_connection_string(cluster: str) -> KustoConnectionStringBuilder:
+    # Todo - Config (Optional): App Id & tenant, and App Key to authenticate with
+    # Todo Learn More: For information on how to procure an AAD Application in Azure see:
+    #  https://docs.microsoft.com/en-us/azure/data-explorer/provision-azure-ad-app
+    app_id = os.environ.get("APP_ID")
+    app_tenant = os.environ.get("APP_TENANT")
+    app_key = os.environ.get("APP_KEY")
+
+    return KustoConnectionStringBuilder.with_aad_application_key_authentication(cluster, app_id, app_key, app_tenant)
+
+
+def create_app_cert_connection_string(cluster: str) -> KustoConnectionStringBuilder:
+    # Todo - Config (Optional): App Id & tenant, and certificate to authenticate with
+    # Todo Learn More: For information on how to procure an AAD Application in Azure see:
+    #  https://docs.microsoft.com/en-us/azure/data-explorer/provision-azure-ad-app
+    app_id = os.environ.get("APP_ID")
+    app_tenant = os.environ.get("APP_TENANT")
+    pem_file_path = os.environ.get("PEM_FILE_PATH")
+    thumbprint = os.environ.get("CERT_THUMBPRINT")
+    public_cert_path = os.environ.get("PUBLIC_CERT_FILE_PATH")  # Only used on Subject Name and Issuer Auth
+    public_certificate = None
+    pem_certificate = None
+
+    try:
+        with open(pem_file_path, "r") as pem_file:
+            pem_certificate = pem_file.read()
+    except Exception as ex:
+        die(f"Failed to load PEM file from {pem_file_path}", ex)
+
+    if public_cert_path is None:
+        try:
+            with open(public_cert_path, "r") as cert_file:
+                public_certificate = cert_file.read()
+        except Exception as ex:
+            die(f"Failed to load public certificate file from {public_cert_path}", ex)
+
+        return KustoConnectionStringBuilder.with_aad_application_certificate_authentication(cluster, app_id, pem_certificate, thumbprint, app_tenant)
+    else:
+        return KustoConnectionStringBuilder.with_aad_application_certificate_sni_authentication(
+            cluster, app_id, pem_certificate, public_certificate, thumbprint, app_tenant
+        )
+
+
+def die(error: str, ex: Exception = None):
+    print(error)
+    if ex is not None:
+        print(ex)
+
+    exit(-1)
+
+
+main()

```

Comment:
usually, this is wrapped with `if __name__ == '__main__'` to ensure it's run as a script and not part of a module